python/cucumber_expressions/argument.py

```diff
@@ -1,42 +1,52 @@
 from __future__ import annotations
 
-from typing import Optional, List
+from typing import Optional, List, Tuple
 
-from cucumber_expressions.group import Group
 from cucumber_expressions.parameter_type import ParameterType
-from cucumber_expressions.tree_regexp import TreeRegexp
+from cucumber_expressions.tree_regexp import TreeRegexp, Group
 from cucumber_expressions.errors import CucumberExpressionError
 
 
 class Argument:
-    def __init__(self, group, parameter_type):
-        self._group: Group = group
-        self.parameter_type: ParameterType = parameter_type
+    def __init__(
+        self, group: Group, parameter_type: ParameterType, name: Optional[str]
+    ):
+        self.group = group
+        self.parameter_type = parameter_type
+        self.name = name
 
     @staticmethod
     def build(
-        tree_regexp: TreeRegexp, text: str, parameter_types: List
+        tree_regexp: TreeRegexp,
+        text: str,
+        parameter_types_and_names: List[Tuple[ParameterType, Optional[str]]],
     ) -> Optional[List[Argument]]:
+        # Check if all elements in parameter_types_and_names are tuples
+        for item in parameter_types_and_names:
+            if not isinstance(item, tuple) or len(item) != 2:
+                raise CucumberExpressionError(
+                    f"Expected a tuple of (ParameterType, Optional[str]), but got {type(item)}: {item}"
```

Comment:
This error is very technical. What should a user do if/when they encounter this error?

Reply:
Good point - I'll have a think - this was mainly done for my benefit when debugging!

Reply:
Tuple is a common type used in Python - the type itself should make sense, but I'll review for clarity

Reply:
Syntax: `{param_name:param_type}`
How do you represent the `Optional[str]` when param_type is not present? With `None`?
For simplicity sake, could it assume `string` type if param_type is omitted?
I think this error is a bit redundant because the parser should handle the errors/exceptions. Meaning that if the code is working properly you will never throw this error.